rasa/telemetry.py

```
@@ -0,0 +1,553 @@
+
+    Do not call this function from outside telemetry.py! This function does not
+    check if telemetry is enabled or not.
+
+    Args:
+        distinct_id: Unique telemetry ID.
+        event_name: Name of the event.
+        properties: Values to report along the event.
+        context: Context information about the event.
+    """
+
+    payload = segment_request_payload(distinct_id, event_name, properties, context)
+
+    if _is_telemetry_debug_enabled():
+        print_telemetry_event(payload)
+        return
+
+    write_key = telemetry_write_key()
+    if not write_key:
+        # If TELEMETRY_WRITE_KEY is empty or `None`, telemetry has not been
+        # enabled for this build (e.g. because it is running from source)
+        logger.debug("Skipping request to external service: telemetry key not set.")
+        return
+
+    headers = segment_request_header(write_key)
+
+    async with aiohttp.ClientSession() as session:
+        async with session.post(
+            SEGMENT_ENDPOINT, headers=headers, json=payload,
+        ) as resp:
+            # handle different failure cases
+            if resp.status != 200:
+                response_text = await resp.text()
+                logger.debug(
+                    f"Segment telemetry request returned a {resp.status} response."
+                    f"Body: {response_text}"
+                )
+            else:
+                data = await resp.json()
+                if not data.get("success"):
+                    logger.debug(
+                        f"Segment telemetry request returned a failure."
+                        f"Response: {data}"
+                    )
+
+
+def _project_hash() -> Text:
+    """Create a hash for the project in the current working directory.
+
+    Returns:
+        project hash
+    """
+    try:
+        remote = check_output(  # skipcq:BAN-B607,BAN-B603
+            ["git", "remote", "get-url", "origin"], stderr=STDOUT
```

Why do we want to capture the error output? This should fail silently, shouldn't it?

---

in examples usually, it gets pipped there if you don't want errors to end up in your error stream. I changed it to pipe it to dev null.